tests/e2e/e2e_upgrade_gov_and_authz_test.go

```diff
@@ -0,0 +1,215 @@
+
+	// fetch authz grants after upgrade
+	authzAfterGrants, err := suite.Kava.Authz.Grants(afterUpgradeCtx, &authz.QueryGrantsRequest{Granter: kavadistModuleAcc, Grantee: govModuleAcc, Pagination: &query.PageRequest{Limit: 1000, CountTotal: true}})
+	suite.Require().NoError(err)
+	suite.Require().Equal(authzAfterGrants.Pagination.Total, uint64(len(authzAfterGrants.Grants)), "expected all grants to have been requested")
+
+	// one kavadist -> gov grants
+	suite.Require().Equal(1, len(authzAfterGrants.Grants))
+
+	grant := authzAfterGrants.Grants[0]
+
+	var authorization authz.Authorization
+	suite.Kava.EncodingConfig.Marshaler.UnpackAny(grant.Authorization, &authorization)
+
+	genericAuthorization, ok := authorization.(*authz.GenericAuthorization)
+	suite.Require().True(ok, "expected generic authorization")
+
+	// kavadist allows gov to MsgSend it's funds
+	suite.Equal(sdk.MsgTypeURL(&banktypes.MsgSend{}), genericAuthorization.Msg)
+	// no expiration
+	var expectedExpiration *time.Time
+	suite.Equal(expectedExpiration, grant.Expiration)
+}
+
+func (suite *IntegrationTestSuite) TestModuleAccountGovTransfers() {
+	suite.SkipIfUpgradeDisabled()
+	suite.SkipIfKvtoolDisabled()
+
+	// the module account (authority) that executes the transfers
+	govAcc := sdk.MustAccAddressFromBech32(govModuleAcc)
+
+	// module accounts for gov transfer test cases
+	communityAcc := sdk.MustAccAddressFromBech32(communityModuleAcc)
+	kavadistAcc := sdk.MustAccAddressFromBech32(kavadistModuleAcc)
+
+	testCases := []struct {
+		name     string
+		sender   sdk.AccAddress
+		receiver sdk.AccAddress
+		amount   sdk.Coin
+	}{
+		{
+			name:     "transfer from community to kavadist for incentive rewards",
+			sender:   communityAcc,
+			receiver: kavadistAcc,
+			amount:   ukava(100e6),
+		},
+		{
+			name:     "transfer from kavadist to community",
+			sender:   kavadistAcc,
+			receiver: communityAcc,
+			amount:   ukava(50e6),
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			// get starting balance of the module account to be funded
+			startingBalance := suite.Kava.QuerySdkForBalances(tc.receiver)
+
+			// create msg exec for transfer between modules
+			msg := banktypes.NewMsgSend(
+				tc.sender,
+				tc.receiver,
+				sdk.NewCoins(tc.amount),
+			)
+			execMsg := authz.NewMsgExec(govAcc, []sdk.Msg{msg})
+
+			// ensure proposal passes
+			suite.submitAndPassProposal([]sdk.Msg{&execMsg})
+
+			// get ending balance of the receiving account
+			endingBalance := suite.Kava.QuerySdkForBalances(tc.receiver)
+			// subtract the starting balance to get the total transferred amount
+			transferredAmount := endingBalance.AmountOf(tc.amount.Denom).Sub(startingBalance.AmountOf(tc.amount.Denom))
+
+			// assert that the transferred amount is greater than or equal to the sent amount
+			// we use >= here since module accounts can accumulate tokens between blocks
+			suite.Require().True(endingBalance.AmountOf(tc.amount.Denom).GTE(transferredAmount))
```

Comment:
Should we be comparing the actual change amount to the msg transfer amount instead of endingBalance?

```suggestion
			suite.Require().True(tc.amount.Amount.GTE(transferredAmount))
```

Reply:
@drklee3 hold off on merge

i ran the tests with
```go
suite.Require().True(transferredAmount.GTE(tc.amount.Amount))
```

ran into this failure:
```
=== RUN   TestIntegrationTestSuite/TestModuleAccountGovTransfers/transfer_from_kavadist_to_community
[whale] 2023/10/31 09:34:36 broadcasting sdk tx. has data = <nil>
[whale] 2023/10/31 09:34:37 response code: 19, hash 9D91BE9F0943C9C78B1028F23E858B3021999A2C83CD9BCC90E006DF2763570E
    e2e_community_update_params_test.go:184: txData.MsgResponses: [&Any{TypeUrl:/cosmos.gov.v1.MsgSubmitProposalResponse,Value:[8 3],XXX_unrecognized:[]}]
[whale] 2023/10/31 09:34:37 broadcasting sdk tx. has data = <nil>
[whale] 2023/10/31 09:34:38 response code: 0, hash DBFBC1DB89C3310EFDA64FB437BF32CC1B703281FFC97BB7BFE3B3E0B262E252
    e2e_upgrade_gov_and_authz_test.go:135: 
                Error Trace:    /kava/kava/tests/e2e/e2e_upgrade_gov_and_authz_test.go:135
                                /go/pkg/mod/github.com/stretchr/testify@v1.8.3/suite/suite.go:112
                Error:          Should be true
```

am looking into it.

Reply:
with additional logging:
```
transferred amount: 49963854
amount: 50000000
```

Reply:
Updating the test to account for this, does seem to be around 35 seconds worth of staking rewards which makes sense with the gov voting period being 30s